galley/pkg/config/analysis/analyzers/injection/injection.go

```diff
@@ -59,9 +60,8 @@ func (a *Analyzer) Analyze(c analysis.Context) {
 
 	c.ForEach(metadata.K8SCoreV1Namespaces, func(r *resource.Entry) bool {
 
-		// Ignore system namespaces
-		// TODO: namespaces can in theory be anything, so we need to make this more configurable
-		if strings.HasPrefix(r.Metadata.Name.String(), "kube-") || strings.HasPrefix(r.Metadata.Name.String(), "istio-") {
+		ns := r.Metadata.Name.String()
+		if util.IsSystemNamespace(ns) {
```

Comment:
The previous behavior skipped istio-system as well. I didn't see that handled in `IsSystemNamespace`.

Reply:
There's a slight behavior change here. Istio now installs with `istio-system` labeled `istio-injection:disabled`, so we aren't going to generate false positives in the normal case and excluding istio-system is redundant. And if someone manually changes that, then I don't feel bad alerting them about it.

The fewer namespace names we need to hard code, the better.